tokio/src/runtime/task/harness.rs

```diff
@@ -67,15 +67,19 @@ where
             }
         };
 
-        // Ensure the task is bound to a scheduler instance. If this is the
-        // first time polling the task, a scheduler instance is pulled from the
-        // local context and assigned to the task.
-        //
-        // The scheduler maintains ownership of the task and responds to `wake`
-        // calls.
-        //
-        // The task reference count has been incremented.
-        self.core().bind_scheduler(self.to_task());
+        if is_not_bound {
+            // Ensure the task is bound to a scheduler instance. If this is the
+            // first time polling the task, a scheduler instance is pulled from the
+            // local context and assigned to the task.
+            //
+            // The scheduler maintains ownership of the task and responds to `wake`
+            // calls.
+            //
+            // The task reference count has been incremented.
+            unsafe {
```

Comment:
Since this is now unsafe, we should probably update the comment to indicate why it's safe to call here?

Reply:
No longer unsafe, but documented regardless. Thanks!